deepspeed/runtime/engine.py

```diff
@@ -2897,12 +2901,16 @@ def _create_zero_checkpoint_files(self, save_dir, tag):
     def _save_checkpoint(self, save_dir, tag, client_state={}):
 
         save_path = self._get_ckpt_name(save_dir, tag)
+
         # A hack to save the checkpointing directory. Pipeline parallelism overrides
         # module_state_dict() and uses this path to save the model. module_state_dict()
-        # then instead just returns None.
+        # then instead just returns None.  The module_state_dict() implementation in
+        # PipelineEngine expects the save path to be set in self._curr_ckpt_path.
         self._curr_ckpt_path = os.path.join(save_dir, tag)
+        module = self.module_state_dict()
+        self._curr_ckpt_path = None
 
-        state = dict(module=self.module_state_dict(),
+        state = dict(module=module,
```

Comment:
What is the significance of of passing `module` instead of `module.state_dict()`?

Reply:
It's been a while, but I think I separated that out mainly to set ``self._curr_ckpt_path`` back to ``None`` immediately after calling ``module_state_dict``.  It looks like the original code was doing that a bit later, though it was actually setting ``self._curr_save_path = None`` instead of ``self._curr_ckpt_path = None``, which I interpreted as a typo in the original code.

Reply:
@adammoody, looks good to me.